File: react-frontend-ts/src/services/MessageService.ts
import axios from 'axios'

const BACKEND_URL = 'http://localhost:8080' 
const GET_MESSAGES_URL = '/demo/messages';
const POST_MESSAGES_URL = '/demo/createmessage';

class MessageService {

    getMessages(){
        return axios.get(BACKEND_URL+GET_MESSAGES_URL);
    }

    createMessage(message: IMessage) {
        return axios.post(BACKEND_URL+POST_MESSAGES_URL, {
            description: message.description,
            time: message.time,
        });
    }
}

export default new MessageService();

File: src/main/java/com/example/demo/service/MessageService.java
package com.example.demo.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import com.example.demo.dao.MessageDao;
import com.example.demo.entities.Message;

/**
 * User Service
 * @author patrice
 *
 */
@Component
public class MessageService {

	@Autowired
	MessageDao messageDao;

	/**
	 * Create a message
	 * @return true if message created, false if message already exist or error
	 * @throws Exception
	 */
	public boolean createMessage(Message message) throws Exception {
	    return messageDao.createMessage(message);
	}
	
	/**
	 * get all messages
	 * @return message List
	 * @throws Exception
	 */
	public List<Message> getMessages() throws Exception {
		return messageDao.getMessages();
	}
	
	
}
	
